spec/p2p/v0.34/addressbook.md

```diff
@@ -0,0 +1,368 @@
+# Address Book
+
+The address book tracks information about other nodes in the network.
+
+The primary information about a node is its peer address,
+composed by a network address and an ID.
+Addresses can be manually configured (e.g., as persistent peers),
+can be learned when a peer attempts to connect to the node,
+or can be retrieved via the [Peer Exchange protocol](./pex-protocol.md).
+
+The [Peer Manager](./peer_manager.md) retrieves from the address book peer
+addresses to dial, so to establish outbound connections.
+It also records in the address book failed attempts to connect to a peer.
+
+The [Peer Exchange protocol](./pex-protocol.md) feds the address book with
+addresses received from peers, and retrieves from the address book random
+selection of addresses it provides to peers.
+
+Once the node is connected to a peer, it uses the address book to record
+information about the quality of the peer.
+The quality of a peer is currently restricted to the definition of
+[good](#good-peers) or [bad](#bad-peers) peers.
+It influences, in particular, the selection of candidate peers to dial.
+
+## New addresses
+
+The `AddAddress` method adds a new address to the address book.
+
+The added address is associated to a *source* address, which identifies the
```

Comment:
Is source address the address of the peer from which new address is learned (not necessarily that peer address)? For example if I learned from peer A address of peer B, then source address for B would be A. right?